spikeinterface/extractors/phykilosortextractors.py

```diff
@@ -97,7 +97,8 @@ def __init__(self, folder_path, exclude_cluster_groups=None, keep_good_only=Fals
             del cluster_info["id"]
 
         # update spike clusters and times values
-        spike_clusters_clean_idxs = np.in1d(spike_clusters, cluster_info["cluster_id"].values).nonzero()[0]
+        bad_clusters = [clust for clust in clust_id if clust not in cluster_info['cluster_id'].values]
+        spike_clusters_clean_idxs = ~np.isin(spike_clusters, bad_clusters)
```

Comment:
Since there are usually less bad clusters than good ones, it's faster to call `np.isin` giving it the bad clusters, and then flipping the boolean array.